### ISSUE TITLE
Huge warning when required modules are inactive [Merge 0.x into master]

### ISSUE THREAD

**Issue by cdfa** [zotonic/zotonic#3045]
## `Zotonic` version
0.x and 1.x

## `OS` version
Irrelevant

## Description
* **Motivation**
We keep finding issues on websites that are caused by some modules being inactive. It happens so often that checking if all the required modules are on should be the first step in our debugging process. Checking this is pretty easy, but still we often forget about it, because this problem can cause such a great variety of issues (where we think something else is going on) and modules can go down at unexpected times (or for example: you still need to start a module that you just added).
In the end, we still spend a significant amount of time debugging issues caused by modules being inactive.

* **Proposal**
To make it a bit easier to notice, I would like a HUGE WARNING (okay, that's exaggerated, but large enough that you can't miss it) to appear in the admin dashboard when one of the required modules is inactive.
Additionally, I would like to have a warning about this in the menu bar, so you can see it from any page.

## Current behavior
Zotonic behaves incorrectly because of required modules being inactive without any warning.

## Expected behavior
Zotonic gives warnings in the admin interface when required modules are inactive.

**Comment by mworrell** [comment]
I think this is mainly a 0.x problem, caused by sites starting in parallel and exhausting resources.  So I will file it for the 0.68 and then afterwards patch it for the 1.x

**Comment by cdfa** [comment]
Great! I don't think it's always caused by sites starting in parallel, though. Sometimes there's confusion when adding new modules or it doesn't start for external reasons/misconfiguration.

**Comment by mworrell** [comment]
Merge change from https://github.com/zotonic/zotonic/pull/3249 into master

**Comment by cdfa** [comment]
Thanks for fixing this!
However, I noticed that there is only a warning when the module is installed, but not running. If there is a module that is required but not installed, it would be nice to get a similar warning. Especially since I did not get any errors in the zotonic console for 2 instances where this happened, while the server was responding with 500 to requests.

Do you think something like that is possible?
Since there is already something merged into 0.x we could make this into a separate issue, but I thought I'd start here since it was still open.

**Comment by mworrell** [comment]
Can you start a new issue?  Then we can track it from there, as this might need more logic during module install regarding what to do when installing a module that is not available.
